### PR TITLE
COMMERCE-11715 - Issue with Custom Fields not being returned in OptionValue API

### PR DESCRIPTION
**Pull request by victorcdp**
Related Issue: [COMMERCE-11715](https://issues.liferay.com/browse/COMMERCE-11715)

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Can you review this one? Tied to an LPP. Thank you!

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Can you see if it's alright now? Thank you.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -763,6 +763,13 @@ components:
                         type: object
                     readOnly: true
                     type: object
+                customFields:
+                    items:
+                        $ref: "#/components/schemas/CustomField"
+                    type: array
+                expando:
+                    additionalProperties: true
+                    type: object
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:772]: this is the old patter, you don't need to add this

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/OptionValueDTOConverter.java [hunk 2]
```diff
@@ -46,9 +48,18 @@ public OptionValue toDTO(DTOConverterContext dtoConverterContext)
...
 		return new OptionValue() {
 			{
 				actions = dtoConverterContext.getActions();
+				customFields = CustomFieldsUtil.toCustomFields(
+					dtoConverterContext.isAcceptAllLanguages(),
+					CPOptionValue.class.getName(),
+					cpOptionValue.getCPOptionValueId(),
+					cpOptionValue.getCompanyId(),
+					dtoConverterContext.getLocale());
+				expando = expandoBridge.getAttributes();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/OptionValueDTOConverter.java:62]: no need of this expando field

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Should be good now. Thank you.

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Pull request closed without merging**